interview-questions.md

```diff
@@ -8,24 +8,38 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating a Rails application with a model called Cohort that has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
-Your answer:
+# Your answer:
+The foreign key is the mapping on the Student model which indicates to which Cohort the student belongs to.
 
-Researched answer:
+# Researched answer:
+Because the foreign key is in the references column of the Student database, a migration to remove the reference and replace it with a correct column or to fix the foreign_key attribute may do the trick.
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+# Your answer:
+I do not know.
 
-Researched answer:
+# Researched answer:
+RESTful routes which are expecting params, must be passed params or rails will throw the appropriate error.
 
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
+# Your answer:
+Rails can generate a model, which creates various folders and files in the rails application, including a migration folder.
+Rails can generate a Controller, which does the same.
 
-Researched answer:
+# Researched answer:
+Rails can also generate a scaffold, which populates the controller around a database with the default RESTFul methods. 
+Rails can also generate a testing environment (?) using RSpec, this must be done before models are generated.
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
+# The action category looks like SQL keywords, covering the CRUD paradigm, displayed in the Rails console when dealing with a generated and migrated model.
+
+# The location category looks like the URL one would type in a Rails app to navigate to wherever the "Students" controller deems appropriate, maybe a default "index" method with a "render html" command, or instance variables which are passed on to an ERB file for display.
+
+# I do not know the name of the controller method called by each route or what action it would perform. 
```

Comment:
This is a crucial concepts to understand in full stack development.  Make sure to check your resources here: [restful routes](https://github.com/learn-academy-2022-golf/Syllabus/blob/main/rails/restful-routes-crud.md)